build-system/tasks/release/index.js

```diff
@@ -0,0 +1,370 @@
+const STATIC_FILES_DIR = path.resolve(__dirname, 'static');
+
+// List of individual files to copy directly from the Git workspace to dist/.
+const STATIC_FILE_PATHS = ['build-system/global-configs/caches.json'];
+
+// List of individual files to copy from post-build files, key-values as the
+// from-to of the copy operations.
+const POST_BUILD_MOVES = {
+  'dist.tools/experiments/experiments.cdn.html': 'dist/experiments.html',
+  'dist.tools/experiments/experiments.js': 'dist/v0/experiments.js',
+  'dist.tools/experiments/experiments.js.map': 'dist/v0/experiments.js.map',
+};
+
+// URL to the JSON info API for the amp-sw package on the npm registry.
+const AMP_SW_NPM_PACKAGE_URL = 'https://registry.npmjs.org/@ampproject/amp-sw';
+
+// List of directories to keep in the temp directory from each flavor's build.
+const DIST_DIRS = ['dist', 'dist.3p', 'dist.tools'];
+
+// Mapping from a channel's RTV prefix, to an object with channel configuration
+// data based on the spec/amp-framework-hosting.md document. The fields are:
+// - type: machine name of the channel (Some of these are the same as their name
+//   in spec/amp-framework-hosting.md, but some are different. Those that have a
+//   different machine name are marked in a comment.)
+// - configBase: name of the config JSON file to prepend as the AMP_CONFIG to
+//   all the entry files in this channel. Either `canary` or `prod` (these files
+//   are located in build-system/global-configs/${configBase}-config.json).
+const CHANNEL_CONFIGS = {
+  '00': {type: 'experimental', configBase: 'canary'},
+  '01': {type: 'production', configBase: 'prod'}, // Spec name: 'stable'
+  '02': {type: 'control', configBase: 'prod'},
+  '03': {type: 'rc', configBase: 'prod'}, // Spec name: 'beta'
+  '04': {type: 'nightly', configBase: 'prod'},
+  '05': {type: 'nightly-control', configBase: 'prod'},
+  '10': {type: 'experimentA', configBase: 'prod'},
+  '11': {type: 'experimentB', configBase: 'prod'},
+  '12': {type: 'experimentC', configBase: 'prod'},
+  '20': {type: 'experimentA-control', configBase: 'prod'}, // Spec name: 'inabox-experimentA-control'
+  '21': {type: 'experimentA', configBase: 'prod'}, // Spec name: 'inabox-experimentA'
+  '22': {type: 'experimentB-control', configBase: 'prod'}, // Spec name: 'inabox-experimentB-control'
+  '23': {type: 'experimentB', configBase: 'prod'}, // Spec name: 'inabox-experimentB'
+  '24': {type: 'experimentC-control', configBase: 'prod'}, // Spec name: 'inabox-experimentC-control'
+  '25': {type: 'experimentC', configBase: 'prod'}, // Spec name: 'inabox-experimentC'
+};
+
+// Mapping of entry file names to a dictionary of AMP_CONFIG additions.
+const TARGETS_TO_CONFIG = MINIFIED_TARGETS.flatMap((minifiedTarget) => [
+  {file: `${minifiedTarget}.js`, config: {}},
+  {file: `${minifiedTarget}.mjs`, config: {esm: 1}},
+]);
+
+function logSeparator_() {
+  log('---\n\n');
+}
+
+async function prepareEnvironment_(outputDir, outputTempDir) {
```

Comment:
High level comment: For maintainability, can you add a short JSDoc summary for each function in this file? (Context: #28387)

Reply:
Done